src/web/mjs/connectors/CxC.mjs

```diff
@@ -0,0 +1,254 @@
+                value: ''
+            },
+            password: {
+                label: 'Password',
+                description: 'Password for login with your CxC account.\nAn account is required to access 18+ content (18禁).\nA fan subscription is required to access VIP content (訂閱粉絲限定).',
+                input: 'password',
+                value: ''
+            }
+        };
+    }
+
+    async _initializeConnector() {
+        await this._getUUID();
+        await this._login();
+    }
+
+    async _getMangaFromURI(uri) {
+        await this._login();
+        const bookAddress = uri.href;
+        const storeName = bookAddress.match(/\/store\/([a-zA-Z0-9_]+)\//)[1];
+        const bookId = bookAddress.match(/\/book\/(\d+)/)[1];
+        let storeRequest = new Request(`${this.api}/store/${storeName}?v=2`, this.requestOptions);
+        let storeData = await this.fetchJSON(storeRequest);
+        const id = `/store/${storeData.data.id}/book/${bookId}`;
+        let bookRequest = new Request(this.api + id, this.requestOptions);
+        let bookData = await this.fetchJSON(bookRequest);
+        if (bookData.data.work_category.id != 1) {
+            throw Error('CxC book type must be comic (漫畫).');
+        }
+        return new Manga(this, id, bookData.data.name);
+    }
+
+    async _getMangasFromPage(page) {
+        await this._login();
+        try {
+            let request = new Request(`${this.api}/book?page=${page}&per_page=${this.booksPerPage}&sort_by=updated_at&work_category=1&lang=&word_count=0`, this.requestOptions);
+            let bookData = await this.fetchJSON(request);
+            bookData = bookData.data.data.filter(book => book.work_category.id == 1);
+            return bookData.map(book => {
+                return {
+                    id: `/store/${book.store.id}/book/${book.id}`,
+                    title: book.name
+                };
+            });
+        } catch (error) {
+            return [];
+        }
+    }
+
+    async _getMangas() {
+        await this._login();
+        let books = [];
+        for (let page = 1, run = true; run; page++) {
+            const booksFromPage = await this._getMangasFromPage(page);
+            booksFromPage.length >= this.booksPerPage ? books.push(...booksFromPage) : run = false;
+        }
+        return books;
+    }
+
+    async _getChapters(manga) {
+        await this._login();
+        let request = new Request(`${this.api}${manga.id}/chapter`, this.requestOptions);
+        let chapterData;
+        try {
+            chapterData = await this.fetchJSON(request);
+        } catch (error) {
+            if (!this.auth.accessToken) {
+                throw Error("Could not get chapters, may be age-restricted (18禁). Add an account to view chapters.");
+            } else {
+                throw Error("Could not get chapters, your account may not be configured to view age-restricted works (18禁).");
+            }
+        }
+        let chapters = [];
```

Comment:
chapters variable is kinda useless. you can directly return the map results
```

return chapterData.map ((chapter, index) => {
 return {
                 id: `${manga.id}/chapter/${chapter.id+1}`,
                title: `#${chapterNum.toString()} - ${chapter.name}`
          }
});

```

Same stuff elsewhere in the file

Reply:
Thanks for the feedback, not used to doing so much logic in return statements.